network/tailscale/tailscale.yaml

```diff
@@ -0,0 +1,73 @@
+name: tailscale
+depends:
+  - service: cri
+  - network:
+    - addresses
+    - connectivity
+    - etcfiles
+container:
+  entrypoint: ./containerboot
+  args: []
+  environment:
+  - PATH=/usr/local/bin:/sbin
+  - TS_SOCKET=/var/run/tailscale/tailscaled.sock
+  - TS_STATE_DIR=/var/lib/tailscale
+  - TS_USERSPACE=false
+  - TS_ACCEPT_DNS=true
+  security:
+    maskedPaths: []
+    readonlyPaths: []
+    writeableRootfs: true
+    writeableSysfs: true
+  mounts:
+  # libs
+  - source: /lib
+    destination: /lib
+    type: bind
+    options:
+      - bind
+      - ro
+  # more libs
+  - source: /usr/lib
+    destination: /usr/lib
+    type: bind
+    options:
+      - bind
+      - ro
+  - source: /sbin
+    destination: /sbin
+    type: bind
+    options:
+      - bind
+      - ro
+  - source: /usr/local/bin
+    destination: /usr/local/bin
+    type: bind
+    options:
+      - bind
+      - ro
+  - source: /dev/net/tun
+    destination: /dev/net/tun
+    type: bind
+    options:
+      - bind
+      - rw
+  - source: /var/lib/tailscale
+    destination: /var/lib/tailscale
+    type: bind
+    options:
+      - bind
+      - rw
+  - source: /var/run/tailscale
+    destination: /var/run/tailscale
+    type: bind
+    options:
+      - bind
+      - rw
+  - source: /etc/resolv.conf
```

Comment:
containers by default gets the host resolv.conf, so does need to be bind mounted?

Reply:
oh, just saw that it's a `rw` mount, that seems tricky, since talos also has `/etc/resolv` as another bind mount. I guess a better option is to tell talos to update the dns entries, but that requires more work though

Reply:
Yeah I think that bind mount has caused me grief, so it's not the best. The main reason was to allow dns resolution. So that you can reference nodes via their hostnames, when connected.

It might be better to make that a user configurable option, e.g in your talos install, specify the Tailscale dns that the extension enables.

*but* that is also race condition, I imagine if the extension isn't up, you lose dns completely. It might be better to not have magicdns in the first part, though people would need to use the Tailscale IPs at the control plane layer.

I might think about this more, see if there is any alternative.

Reply:
Completely removed resolve.conf binding. Magic DNS needs more design to integrate with talos